### PR TITLE
update autograding tests

### PR DESCRIPTION
**Pull request by lialittis**

### CODE REVIEW

File: tests/Makefile [hunk 1]
```diff
@@ -30,6 +30,6 @@ check: all
 	rm -f mandelbrot_driver_output
 	python3 ./test_mandelbrot.py 0.98
 	bash test_existence_perf.sh
-	bash test_matrix_stat.sh 0.5 0.2 0.4 0.7 0.2
-	bash test_mandelbrot_stat.sh 0.4 0.4 0.4 0.4
+	python3 test_matrix_stat.py 0.5 0.2 0.4 0.1 0.2
```
Comment (taugoust) [inline tests/Makefile:33]: Why did you change the ratios here?

File: .github/classroom/autograding.json [hunk 2]
```diff
@@ -83,7 +83,7 @@
     {
       "name": "Test matrix multiplication perf usage",
       "setup": "",
-      "run": "timeout -s9 2m bash tests/test_matrix_stat.sh 0.5 0.2 0.4 0.7 0.2",
+      "run": "timeout -s9 2m python3 tests/test_matrix_stat.py 0.5 0.2 0.4 0.1 0.2",
```
Comment (taugoust) [inline .github/classroom/autograding.json:86]: Why did you change the ratios here?

File: tests/test_matrix_stat.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    sys.exit(1)
+
+basic_time = extract_value(basic_stat_path, "cpu-clock")
+optimized_time = extract_value(optimized_stat_path, "cpu-clock")
+
+if basic_time * ratios["cpu-clock"] < optimized_time:
+    print("Error: too much time consumed")
+    sys.exit(1)
+
+sys.exit(0)
```
Comment (taugoust) [inline tests/test_matrix_stat.py:98]: I'd say the best would be to have:

`if __name__ == "__main__":`

File: tests/Makefile (hunk 1 above)
Comment (lialittis) [reply tests/Makefile:33]: I double check the ratios in the reference solution repo, I found I made a mistake. It should be much less than 0.1 for the l1 cache misses.

File: .github/classroom/autograding.json (hunk 2 above)
Comment (lialittis) [reply .github/classroom/autograding.json:86]: Same reason as before. For the l1 cache misses, the boundary shoule be much stricter.

**Review by taugoust — APPROVED** [review]
It works on my system, I've run it directly and then also stress-tested it briefly.
